### ISSUE TITLE
Torch.log2 + Torch.round() different and strange behaviour

### ISSUE THREAD

**Issue by creotiv** [pytorch/pytorch#28284]
## 🐛 Bug

During work with Mask-RCNN found such line '2**torch.tensor(approx_scale).log2().round()' which seemed strange to me due to scales<0 and round should round them to int

Different behaviour:

```
2**torch.tensor(0.123,dtype=torch.float64).log2() = 0.123
2**torch.tensor(0.123,dtype=torch.float64).log2().round() = 0.125
torch.tensor(0.123,dtype=torch.float64).round() = 0.
torch.log2(2**torch.tensor(0.123,dtype=torch.float64)).round() =0.
```

Also i didn't find anything about that in Docs.

PyTorch version: 1.2.0
GPU: Active
Environment: Google Colab GPU instance

**Comment by fmassa** [comment]
Here is what you are missing.

Because `0 < scales < 1`, `scales.log2() < 0`. For example
```python
torch.tensor(0.123).log2()
# tensor(-3.0233)
# then applying round, this becomes
# tensor(-3.)
```

**Comment by creotiv** [comment]
@fmassa 
log2(2**x) == x

so 2**torch.tensor(0.123).log2() == 0.123
and torch.tensor(0.123).round() == 0. But 2**torch.tensor(0.123).log2().round() == 0.125

**Comment by fmassa** [comment]
Yes, because we are first rounding in the `log2` space, and then we do the power computation.
The code is equivalent to
```python
2 ** (torch.tensor(0.123).log2().round()) == 0.125
```
If you want to get 0, just change the location of the parenthesis
```python
(2 ** torch.tensor(0.123).log2()).round() == tensor(0.)
```

**Comment by creotiv** [comment]
@fmassa ok then why

`2 ** (torch.tensor(0.123).log2().round()) == 2 ** torch.tensor(0.123).log2().round() == 0.125`
and
`2**torch.tensor(0.123).log2() == (2**torch.tensor(0.123)).log2() == 0.123`
and 
`(2**torch.tensor(0.123)).log2().round() == 0.`

There is definitely some problem with operation pipeline

**Comment by albanD** [comment]
I don't see the issue here?

In the first case, the parenthesis don't change anything. So all three are equal.

In the second case, you write `2^(log2(0.123)) == log2(2^0.123) == 0.123`

And the last one `round(log2(2^0.123)) == 0`

**Comment by creotiv** [comment]
@fmassa @albanD thanks guys.
